quarkus-extension/engine/deployment/src/test/java/org/camunda/bpm/quarkus/engine/test/id/ConfigureDbIdGeneratorTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.quarkus.engine.test.id;
+
+import io.quarkus.test.QuarkusUnitTest;
+import org.camunda.bpm.engine.TaskService;
+import org.camunda.bpm.engine.impl.db.DbIdGenerator;
+import org.camunda.bpm.engine.task.Task;
+import org.camunda.bpm.quarkus.engine.extension.impl.QuarkusProcessEngineConfiguration;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConfigureDbIdGeneratorTest {
+
+  @RegisterExtension
+  static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+      .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+
+  @Inject
+  public TaskService taskService;
+
+  @ApplicationScoped
+  public static class EngineConfigurer {
+
+    @Produces
+    public QuarkusProcessEngineConfiguration engineConfig() {
+      QuarkusProcessEngineConfiguration engineConfig = new QuarkusProcessEngineConfiguration();
+      engineConfig.setIdGenerator(null);
+      return engineConfig;
+    }
+
+  }
+
+  @Test
+  public void shouldConfigureDbIdGenerator() {
```

Comment:
Maybe we can have another test case here that asserts that the `idGenerator` instance is of type `DbIdGenerator`?

Reply:
Sure 👍